calyx/src/analysis/live_range_analysis.rs

```diff
@@ -255,75 +273,196 @@ impl LiveRangeAnalysis {
             ..Default::default()
         };
 
-        build_live_ranges(
+        // Give each control statement a unique "NODE_ID" attribute.
+        ControlId::compute_unique_ids(control, 0, false);
+
+        ranges.build_live_ranges(
             control,
             Prop::default(),
             Prop::default(),
             Prop::default(),
-            &mut ranges,
         );
 
-        //adds (non-state) shareable cells as live in the group they're contained in
-        //we already added (non-state) shareable cells as live in the invoke
-        //they're contained in in build_live_ranges().
-        comp.groups.iter().for_each(|group| {
-            ranges.add_shareable_ranges(
-                &group.borrow().assignments,
-                group.borrow().name(),
-            );
-        });
+        for (node, cells_by_type) in &ranges.invokes_enables_map {
+            if let Some(prop) = ranges.live.get_mut(node) {
+                prop.or_set_in_place(cells_by_type.clone());
+            }
+        }
 
         // Caleb: Right now we run remove-comb-groups before this is used so this code
         // doesn't do anything. Eventually, though, we want to be able to make
-        // remove-comb-groups optional so I will keep this code.
-        comp.comb_groups.iter().for_each(|group| {
-            ranges.add_shareable_ranges(
-                &group.borrow().assignments,
-                group.borrow().name(),
-            )
-        });
+        // remove-comb-groups optional.
```

Comment:
Leave a note here saying that we don't handle comb groups, or better yet, throw an Error message if we see any